### ISSUE TITLE
All windows of RAFCON should be minimizable and restorable at once

### ISSUE THREAD

**Issue by franzlst** [DLR-RM/RAFCON#136]
The undocking of the library tree or the state editor produces new tabs, which makes it uncomfortable to minimize and restore rafcon at once.

> enhancement

Originally created by Benno  at 2016-06-10 10:58:35+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
And with tabs you mean windows in the task bar?

I would say that this in not avoidable, but if you have a solution for this, it would be welcome!


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-06-10 12:46:59+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Please notice commit 93104f008ac8e44eba6acfc9c336f330bd28a684
Undocked windows wont create new windows in task bar anymore. If this is a suitable solution for you, please close this issue.


Originally created by Benno  at 2016-08-26 09:01:22+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
This is definitely very helpful. The only thing left, which I would like to see (if this is possible!) is to minimize the undocked sidebars with the main window.
There is also a problem with the logger sidebar: It does not get its own task bar entry (which is correct), but it does not appear when the main window is focused.


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-08-29 11:02:22+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
You are right, I forgot the logger sidebar completely.
You probably tested the minimizing only with the logger sidebar? The other sidebars are minimizing together with the main window. ;)
I will update the logger sidebar asap.


Originally created by Benno  at 2016-08-29 12:00:03+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Fixed in 5dcf9fb.
Added same functionallity for forgotten logger sidebar. 


Originally created by Benno  at 2016-08-29 12:19:50+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
OK the sidebars are minimizing with the main window. However, they do not reappear with the main window. If the main window in partially hidden and I click on the visible region, the main window and sidebars are arranged to the top. Yet, if I minimize the main window (and with that the sidebars) and I click on the RAFCON entry in the task bar, only the main window appears, not the sidebars. Only if I maximize the main window, the sidebars reappear.


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-08-29 14:36:10+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Unfortunately I could not reconstruct that problem, but i figured out that if I minimize the maximized main window, the sidebars wont disappear.
Please check wether commit 7b63049 fixed those problems.


Originally created by Benno  at 2016-08-29 14:50:24+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Unfortunately, it does not. Can you point me to the line of code in which the sidebars are commanded to reappear? Or is that handles by the OS?


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-08-29 14:58:24+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
I am running sled11 64bit and the sidebars minimize and popup together with the main window. Seems to be an OS issue.
Another problem related to this is, that I cannot change the size of the main window. It is always in full screen mode. Only if I delete the runtime_config then I am able to change the size of the main window, but only until I close RAFCON. Starting RAFCON leads again to the behavior that I cannot minimize the main window.


Originally created by @sebastian-brunner  (Sebastian.Brunner@dlr.de) at 2016-08-30 06:18:32+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Hmm, this is something I cannot reproduce... (openSUSE Leap with Gnome3)


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-08-30 06:50:12+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
In commit 128bd0f I did another small change.
Unfortunally this is an issue which is difficult for me handle since everything seems to work fine for me...

The following code are the callbacks which handle the appearing and disappearing:

```
def bring_undock_window_to_top_callback(self, widget, event, undocked_window):
    undocked_window.present()

def undock_window_callback(self, widget, event, undocked_window):
    if event.changed_mask == gtk.gdk.WINDOW_STATE_ICONIFIED:
        undocked_window.iconify()
    else:
        undocked_window.deiconify()
```

( /source/main_window.py , line 428)


Originally created by Benno  at 2016-08-30 07:29:26+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
My problems are solved with commit 4156f955e49fd303934494c4666bf0daca04636f


Originally created by @franzlst  (Franz.Steinmetz@dlr.de) at 2016-08-30 13:24:16+00:00 (moved from RMC internal repository)

**Comment by franzlst** [comment]
Undocked windows cant be miximized anymore and lose their position after following procedure:
1. undock window
2. move to prefered position
3. dock again 
4. undock again
5. minimize main window
6. open main window


Originally created by Benno  at 2016-08-31 10:41:44+00:00 (moved from RMC internal repository)
